File: Sources/AdbShell/Models/Device.swift
import Foundation

struct Device: Identifiable, Hashable {
    enum State: String {
        case device
        case offline
        case unauthorized
        case noPermissions = "no permissions"
        case unknown

        var label: String {
            switch self {
            case .device: return L("device.state.connected")
            case .offline: return L("device.state.offline")
            case .unauthorized: return L("device.state.unauthorized")
            case .noPermissions: return L("device.state.noPermissions")
            case .unknown: return L("device.state.unknown")
            }
        }

        var isReady: Bool { self == .device }
    }

    var id: String { serial }
    let serial: String
    let state: State
    var model: String?
    var product: String?
    var transportId: String?
    var isNetwork: Bool { serial.contains(":") }

    var displayName: String {
        model?.replacingOccurrences(of: "_", with: " ") ?? serial
    }
}

File: desktop/src/main/adb/types/Device.ts
// Порт Sources/AdbShell/Models/Device.swift

export type DeviceState = 'device' | 'offline' | 'unauthorized' | 'noPermissions' | 'unknown';

export interface Device {
  serial: string;
  state: DeviceState;
  model?: string;
  product?: string;
  transportId?: string;
}

export function isNetworkDevice(device: Device): boolean {
  return device.serial.includes(':');
}

export function isReadyState(state: DeviceState): boolean {
  return state === 'device';
}

/** Имя для отображения — модель (с "_" заменённым на пробел) или serial, если модели нет. */
export function displayName(device: Device): string {
  return device.model ? device.model.replace(/_/g, ' ') : device.serial;
}
